### PR TITLE
TypeScript migration for auth components

### PR DESCRIPTION
**Pull request by gsouquet**

### CODE REVIEW

File: src/async-components/views/dialogs/eventindex/DisableEventIndexDialog.tsx [hunk 1]
```diff
@@ -16,7 +16,6 @@ limitations under the License.
 
 import React from 'react';
 import * as sdk from '../../../../index';
```
Comment (SimonBrandner) [inline src/async-components/views/dialogs/eventindex/DisableEventIndexDialog.tsx:18]: Could we please get rid of these while migrating if possible?

**Review by SimonBrandner — COMMENTED** [review]
Hopefully, it's possible and won't result in test failures

File: src/@types/global.d.ts [hunk 2]
```diff
@@ -90,6 +90,9 @@ declare global {
         mxUIStore: UIStore;
         mxSetupEncryptionStore?: SetupEncryptionStore;
         mxRoomScrollStateStore?: RoomScrollStateStore;
+        grecaptcha: any;
+        // eslint-disable-next-line
+        mx_on_recaptcha_loaded: () => void;
```
Comment (t3chguy) [inline src/@types/global.d.ts:95]: Blocking on https://github.com/matrix-org/matrix-react-sdk/pull/6401 which improves this area

File: src/components/views/auth/CountryDropdown.tsx [hunk 3]
```diff
@@ -41,15 +39,26 @@ function countryMatchesSearchQuery(query, country) {
     return false;
 }
 
+interface IProps {
+    value?: string;
+    onOptionChange: (country: PhoneNumberCountryDefinition) => void;
+    isSmall: boolean;
```
Comment (t3chguy) [inline src/components/views/auth/CountryDropdown.tsx:45]:
```suggestion
    isSmall: boolean; // if isSmall, show +44 in the selected value
```

Would be good to keep the comment that was here, given that `isSmall` isn't nearly descriptive enough

**Review by t3chguy — CHANGES_REQUESTED** [review]

**Comment by t3chguy** [comment]
Also thanks @SimonBrandner for first round of review

I'd not bother resolving the conflicts just yet, there's about to be another one when the related PR lands

**Comment by SimonBrandner** [comment]
> Also thanks @SimonBrandner for first round of review

I wouldn't really call it a review - I just couldn't help myself, `sdk.getComponent`s are painful to look at :smile:

File: src/@types/global.d.ts [hunk 4]
```diff
@@ -92,6 +92,7 @@ declare global {
         mxUIStore: UIStore;
         mxSetupEncryptionStore?: SetupEncryptionStore;
         mxRoomScrollStateStore?: RoomScrollStateStore;
+        grecaptcha: any;
```
Comment (t3chguy) [inline src/@types/global.d.ts:95]: This is now specified lower and shouldn't be needed anymore https://github.com/matrix-org/matrix-react-sdk/pull/6412/files#diff-af565283322a1965b4ea9bfb48fe1f578631a934aadbfc7a413fe0d5e84746fbR194

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

:shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit:

**Pull request merged**